app/javascript/components/bootcamp/SolveExercisePage/TestResultsView/TestResultsButtons.tsx

```diff
@@ -1,19 +1,63 @@
-import React from 'react'
+import React, { useCallback, useEffect, useMemo } from 'react'
 import { assembleClassNames } from '@/utils/assemble-classnames'
 import useTestStore from '../store/testStore'
 import useEditorStore from '../store/editorStore'
 import type { InformationWidgetData } from '../CodeMirror/extensions/end-line-information/line-information'
 import { useShouldAnimate } from './useShouldAnimate'
+import useTaskStore from '../store/taskStore/taskStore'
+import useAnimationTimelineStore from '../store/animationTimelineStore'
 
 const TRANSITION_DELAY = 0.1
 
 export function TestResultsButtons() {
   const { testSuiteResult, setInspectedTestResult, inspectedTestResult } =
     useTestStore()
-  const { setInformationWidgetData } = useEditorStore()
-
+  const { setInformationWidgetData, setReadonly } = useEditorStore()
+  const { wasFinishLessonModalShown } = useTaskStore()
+  const { shouldAutoplayAnimation } = useAnimationTimelineStore()
   const { shouldAnimate } = useShouldAnimate(testSuiteResult)
 
+  const isSpotlightActive = useMemo(() => {
+    if (!testSuiteResult) return false
+    return !wasFinishLessonModalShown && testSuiteResult.status === 'pass'
```

Comment:
Same comment as above.

Reply:
Should this be deduped somehow? Or is it ok to have it twice. Feels maybe a little risky to have the same condition in two places?